### PR TITLE
Checking if the 2d canvas used in webgl texture source change pixel values

### PR DESCRIPTION
**Pull request by xlai-GitHub**
I am adding an additional check to existing webgl canvas tests to check whether image data of a 2d canvas change pixel values when used as webgl texture source. The test was written because of an issue reported by users and reproducible by some developers on Mac devices in previous Chrome versions (https://code.google.com/p/chromium/issues/detail?id=446380). Similar changes have already been landed in the corresponding webgl tests in Chromium (https://chromium.googlesource.com/chromium/src/+/190da73405a8a3393d8b96f423e961326de1c7b3).

### CODE REVIEW

File: conformance-suites/1.0.2/conformance/resources/tex-image-and-sub-image-2d-with-canvas.js [hunk 1]
```diff
@@ -83,6 +83,20 @@ function generateTest(pixelFormat, pixelType, prologue) {
       setCanvasToRedGreen(ctx);
     }
 
+    function checkSourceCanvasImageData(imageDataBefore, imageDataAfter) {
```
Comment (kenrussell) [inline conformance-suites/1.0.2/conformance/resources/tex-image-and-sub-image-2d-with-canvas.js]: The working group only updates the earlier conformance suite snapshots in order to fix bugs, and this is adding new tests. Could you please undo this change? Thanks.

File: conformance-suites/1.0.3/conformance/resources/tex-image-and-sub-image-2d-with-canvas.js [hunk 2]
```diff
@@ -80,6 +80,20 @@ function generateTest(pixelFormat, pixelType, prologue) {
       ctx.canvas.height = 2;
       setCanvasToRedGreen(ctx);
     }
+    
+    function checkSourceCanvasImageData(imageDataBefore, imageDataAfter) {
```
Comment (kenrussell) [inline conformance-suites/1.0.3/conformance/resources/tex-image-and-sub-image-2d-with-canvas.js]: Please undo the changes in this file too. Thanks.

**Comment by kenrussell** [comment]
The changes to the top of tree file (sdk/tests/js/tests/tex-image-and-sub-image-2d-with-canvas.js) look fine. Could you please undo the other ones, squash commits and update this pull request? Please see https://www.khronos.org/webgl/wiki/Using_Github_To_Contribute for instructions. Thanks.

(Please comment here once you've updated the pull request.)

**Comment by xlai-GitHub** [comment]
Have undone the other two files accordingly.

**Comment by kenrussell** [comment]
Thanks. Looks good.
